src/backend/distributed/utils/placement_connection.c

```diff
@@ -0,0 +1,574 @@
+/*-------------------------------------------------------------------------
+ *
+ * placement_connection.c
+ *   Per placement connection handling.
+ *
+ * Copyright (c) 2016-2017, Citus Data, Inc.
+ *
+ *-------------------------------------------------------------------------
+ */
+
+
+#include "postgres.h"
+
+#include "distributed/connection_management.h"
+#include "distributed/hash_helpers.h"
+#include "distributed/metadata_cache.h"
+#include "distributed/placement_connection.h"
+#include "utils/hsearch.h"
+#include "utils/memutils.h"
+
+
+/* information about a connection reference to a placement */
+typedef struct ConnectionReference
+{
+	/* identity information about the connection */
+	uint64 shardId;
+	uint64 placementId;
+	const char *userName;
```

Comment:
Worth adding a note, e.g. "Stores the user as which we read/modified the placement. We must not reuse connections that were performed using a different role, since it would not have the right permissions."